File: 83-remove-duplicates-from-sorted-list/83-remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }
        
        ListNode curr = head.next;
        ListNode prev = head;
        
        while(curr != null) {
            if (prev.val != curr.val) {
                prev = curr;
            } else {
                prev.next = curr.next;
            }
            
            curr = curr.next;
        }
        
        return head;
    }
}

File: 83-remove-duplicates-from-sorted-list/83-remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head == None or head.next == None:
            return head
        
        curr = head.next;
        prev = head;
        
        while curr is not None:
            if prev.val != curr.val:
                prev = curr
            else:
                prev.next = curr.next
            curr = curr.next
        
        return head